File: GarbageV5/app/src/main/java/dk/itu/garbagev5/ItemsDB.java
package dk.itu.garbagev5;

import java.util.HashMap;

public class ItemsDB {

    private static ItemsDB sItemsDB;
    private final HashMap<String, String> itemsMap = new HashMap<>();

    private ItemsDB() { fillItemsDB(); }
    public static ItemsDB get() {
        if (sItemsDB == null) sItemsDB = new ItemsDB();
        return sItemsDB;
    }

    // search for category
    public String findCategory(String itemName) {
        String category = itemsMap.get(itemName.toLowerCase());
        if (category != null) {
            return itemName + " should be placed in: " + category;//
        }
        return itemName + " should be placed in: not found";
    }

    // add new items
    public void addItem(String what, String where) {
        itemsMap.put(what.toLowerCase(), where.toLowerCase());
    }

    // Item list
    public String listItems(){
        String r = "";
        for (HashMap.Entry <String, String> item: itemsMap.entrySet())
            r = r + "\n "+item.getKey() + " in "+item.getValue();
        return r;
    }

    private void fillItemsDB(){
        itemsMap.put("newspaper","Paper");
        itemsMap.put("can","Metal");
        itemsMap.put("magazine", "Paper");
        itemsMap.put("milk carton","Food");
        itemsMap.put("stone box", "Cardboard");
    }

}

File: GarbageV7/app/src/main/java/dk/itu/garbagev7/ItemsDB.kt
package dk.itu.garbagev7

class ItemsDB private constructor() {
    private val itemsMap: HashMap<String, String> = LinkedHashMap()

    init{
        fillItemsDB()
    }

    companion object {
        private var sItemsDB: ItemsDB? = null

        fun get(): ItemsDB {
            return sItemsDB ?: ItemsDB().also { sItemsDB = it }
        }
    }

    fun findCategory(itemName: String): String {
        val category = itemsMap[itemName.lowercase()]
        return if (category != null) {
            "$itemName should be placed in: $category"
        } else {
            "$itemName should be placed in: not found "
        }
    }



    fun addItem(what: String, where: String) {
        val whereU = where.trim().lowercase().replaceFirstChar {
            it.titlecase()
        }
        itemsMap[what.lowercase()] = whereU
    }

    fun deleteItem(itemName: String){
        itemsMap.remove(itemName.lowercase())
    }

    fun listItems():String{
        return itemsMap.entries.joinToString("\n") { (key, value) -> "$key in $value" }
    }

    private fun fillItemsDB() {
        itemsMap["newspaper"] = "Paper"
        itemsMap["can"] = "Metal"
        itemsMap["magazine"] = "Paper"
        itemsMap["milk carton"] = "Food"
        itemsMap["stone box"] = "Cardboard"
    }

    fun getItemList(): List<Item> {
        return itemsMap.map { (key, value) -> Item(key, value) }
    }

    fun size(): Int {
        return itemsMap.size
    }


}